### PR TITLE
very important top priority virus bug fix

### PR DESCRIPTION
**Pull request by Arthurdentist**
Skeletons no longer get itchy skin from the 'itching syndrome' in virology.

### CODE REVIEW

**Comment by gbasood** [comment]
Don't we have helpers defined for those typechecks?

**Comment by ghost** [comment]
best PR 2016

**Comment by Intigracy** [comment]
https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/__HELPERS/_macros.dm#L20

As-is this isn't going to work because not every skeleton player is human/skeleton, only admin spawned mobs are.

**Comment by gbasood** [comment]
Yeah, there should be a helper for vox skellies.

**Comment by Intigracy** [comment]
There's not, but iirc they're subtypes of skeletons anyway.

**Comment by Arthurdentist** [comment]
skeletal vox aren't a sub-type of skellingtons believe it or not

**Comment by Intigracy** [comment]
yes they are

/datum/species/skellington/skelevox

**Comment by Intigracy** [comment]
Currently the way species work is that you're spawned in as a flat human and then assigned the species (except in the case of admin spawned human/skeleton or whatever mobs), your check doesn't work as-is.

File: code/modules/virus2/effect.dm [hunk 1]
```diff
@@ -861,7 +861,10 @@ var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/mon
 	name = "Itching"
 	stage = 1
 /datum/disease2/effect/itching/activate(var/mob/living/carbon/mob,var/multiplier)
-	to_chat(mob, "<span class='warning'>Your skin itches!</span>")
+	if(istype(mob, /mob/living/carbon/human/skellington)||istype(mob, /mob/living/carbon/human/skelevox))
```
Comment (Intigracy) [inline code/modules/virus2/effect.dm]: just gotta do if(isskellington(mob)) here

**Review by Intigracy — DISMISSED** [review]

**Comment by Shadowmech88** [comment]
Why not just check for the NO_SKIN species flag?

If a species has no skin, its skin shouldn't itch.

**Comment by Intigracy** [comment]
Or that.

File: code/modules/virus2/effect.dm (hunk 1 above)
Comment (clusterfack) [inline code/modules/virus2/effect.dm]: Should be checking species instead

Comment (Intigracy) [reply code/modules/virus2/effect.dm]: isskellington(mob) is a species checking helper

Comment (gbasood) [reply code/modules/virus2/effect.dm]: pretty sure that check doesnt check for skelevox but yes

**Comment by gbasood** [comment]
@Intigracy 

> yes they are
> /datum/species/skellington/skelevox

He's checking the mob not the species fuck
Using only isskellington would break it because the mob itself is not a child of skeleton

**Comment by Shadowmech88** [comment]
> isskellington would break it because the mob itself is not a child of skeleton

isskellington() checks whether the mob is human and then checks whether it has a species var and, if so, checks istype(mob.species, /datum/species/skellington)

It doesn't directly check if the mob is a /mob/living/carbon/human/skellington.

**Comment by Intigracy** [comment]
/mob/living/carbon/human/skellington also only exists via adminspawn

Please read all of my comments. As-is, this code DOES NOT WORK.

**Comment by Intigracy** [comment]
fix when

feel free to actually test it on a human joining as a skeleton if for whatever reason you don't believe me

**Comment by Arthurdentist** [comment]
This compiles but I haven't had time to test it.

Will do that tomorrow.

File: code/modules/virus2/effect.dm [hunk 2]
```diff
@@ -861,7 +861,12 @@ var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/mon
 	name = "Itching"
 	stage = 1
 /datum/disease2/effect/itching/activate(var/mob/living/carbon/mob,var/multiplier)
-	to_chat(mob, "<span class='warning'>Your skin itches!</span>")
+	var/mob/living/carbon/human/H = mob
+	if (istype(H) && H.species.flags & NO_SKIN)
```
Comment (Exxion) [inline code/modules/virus2/effect.dm]: `if(istype(H) && H.species && H.species.flags & NO_SKIN)`

**Review by Exxion — DISMISSED** [review]

File: code/modules/virus2/effect.dm (hunk 2 above)
Comment (Exxion) [reply code/modules/virus2/effect.dm]: Or, if you want to be SUPER extra careful, you could replace the second check there with `istype(H.species)`

**Review by ComicIronic — APPROVED** [review]

**Pull request merged**